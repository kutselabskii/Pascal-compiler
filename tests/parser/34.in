program abc;

var a, i: integer;

begin
if (a > 0) then
	if (a < 5) then
		a := 5
else
	a := 3;
end.