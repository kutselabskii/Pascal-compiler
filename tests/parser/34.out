|__ program
    |-- abc //Program_heading
    |__ //Block
        |-- //Declaration part
        |   |-- var
        |   |   |-- a
        |   |   |   |__  : integer
        |   |   |__ i
        |   |       |__  : integer
        |__ //Statement part
            |__ //Statement sequence
                |-- if
                |   |-- strictly_greater_than
                |   |   |-- a
                |   |   |   |__  : integer
                |   |   |__ 0
                |   |-- if
                |   |   |-- strictly_less_than
                |   |   |   |-- a
                |   |   |   |   |__  : integer
                |   |   |   |__ 5
                |   |   |-- //Assignment_statement
                |   |   |   |-- a
                |   |   |   |   |__  : integer
                |   |   |   |__ 5
                |   |   |__ //Assignment_statement
                |   |       |-- a
                |   |       |   |__  : integer
                |   |       |__ 3
